catalog/core/catalog-core-defaultvalues/src/test/java/ddf/catalog/data/defaultvalues/DefaultAttributeValueImplRegistryImplTest.java

```diff
@@ -0,0 +1,263 @@
+        registry.addDefaultValue(new DefaultAttributeValueImpl(BASIC_METACARD_NAME,
+                TITLE,
+                DEFAULT_1), null);
+        registry.addDefaultValue(new DefaultAttributeValueImpl(OTHER_METACARD_NAME,
+                TITLE, DEFAULT_2), null);
+
+        verifyRegistryDefaultValue(BASIC_METACARD_NAME, TITLE, DEFAULT_1);
+        verifyRegistryDefaultValue(OTHER_METACARD_NAME, TITLE, DEFAULT_2);
+    }
+
+    @Test
+    public void testOverwriteDefaultValue() {
+        registry.addDefaultValue(new DefaultAttributeValueImpl(BASIC_METACARD_NAME,
+                TITLE,
+                DEFAULT_1), null);
+        registry.addDefaultValue(new DefaultAttributeValueImpl(BASIC_METACARD_NAME,
+                TITLE,
+                DEFAULT_2), null);
+        verifyRegistryDefaultValue(BASIC_METACARD_NAME, TITLE, DEFAULT_2);
+    }
+
+    @Test
+    public void testGetUnregisteredMetacardType() {
+        verifyRegistryDefaultValueNotPresent(BASIC_METACARD_NAME, TITLE);
+    }
+
+    @Test
+    public void testGetUnregisteredAttribute() {
+        registry.addDefaultValue(new DefaultAttributeValueImpl(BASIC_METACARD_NAME,
+                POINT_OF_CONTACT,
+                DEFAULT_1), null);
+        verifyRegistryDefaultValueNotPresent(BASIC_METACARD_NAME, TITLE);
+    }
+
+    @Test
+    public void testRemoveSingleDefaultValue() {
+        registry.addDefaultValue(new DefaultAttributeValueImpl(TITLE, DEFAULT_1), null);
+        registry.addDefaultValue(new DefaultAttributeValueImpl(BASIC_METACARD_NAME,
+                TITLE,
+                DEFAULT_1), null);
+        registry.addDefaultValue(new DefaultAttributeValueImpl(BASIC_METACARD_NAME,
+                POINT_OF_CONTACT,
+                DEFAULT_2), null);
+
+        registry.removeDefaultValue(new DefaultAttributeValueImpl(BASIC_METACARD_NAME,
+                TITLE,
+                DEFAULT_1), null);
+        registry.removeDefaultValue(new DefaultAttributeValueImpl(TITLE, DEFAULT_1), null);
+
+        verifyRegistryDefaultValue(BASIC_METACARD_NAME, POINT_OF_CONTACT, DEFAULT_2);
+        verifyRegistryDefaultValueNotPresent(BASIC_METACARD_NAME, TITLE);
+        verifyRegistryDefaultValueNotPresent(TITLE);
+    }
+
+    @Test
+    public void testGlobalDefaults() {
+        registry.addDefaultValue(new DefaultAttributeValueImpl(TITLE, DEFAULT_1), null);
+        registry.addDefaultValue(new DefaultAttributeValueImpl(BASIC_METACARD_NAME,
+                TITLE,
+                DEFAULT_3), null);
+        registry.addDefaultValue(new DefaultAttributeValueImpl(POINT_OF_CONTACT, DEFAULT_2), null);
+
+        verifyRegistryDefaultValue(BASIC_METACARD_NAME, TITLE, DEFAULT_3);
+        verifyRegistryDefaultValue(OTHER_METACARD_NAME, TITLE, DEFAULT_1);
+        verifyRegistryDefaultValue(BASIC_METACARD_NAME, POINT_OF_CONTACT, DEFAULT_2);
+        verifyRegistryDefaultValue(OTHER_METACARD_NAME, POINT_OF_CONTACT, DEFAULT_2);
+    }
+
```

Comment:
You can add a test here which tests global fallback a metacard's specific default is removed